### PR TITLE
Split hawk/dove simulation into two variants for single and variable risk attitudes

### PR DESCRIPTION
**Pull request by rlskoeser**

### CODE REVIEW

**Comment by rlskoeser** [comment]
@quadrismegistus I've successfully split out the hawk/dove model and agent code so I can separate single and variable risk attitudes, which are increasingly diverging in terms of parameters. I think everything is working, but would appreciate if you could give a quick look over to see if there's anything that doesn't make sense or that I might have not thought about.

File: simulatingrisk/batch_run.py [hunk 1]
```diff
@@ -47,46 +48,50 @@ def riskyfood_batch_run(args=None):
...
-            "risk_attitudes": "variable",
-        }
-        iterations = 5
-    elif args.risk_attitudes == "single":
-        params = {
-            "grid_size": 20,
-            "risk_attitudes": "single",
-            "agent_risk_level": [0, 1, 2, 3, 4, 5, 6, 7, 8],
-        }
-        iterations = 1
```
Comment (quadrismegistus) [inline simulatingrisk/batch_run.py:70]: It's cool that these params can be stored elsewhere via the subclassed models

File: simulatingrisk/hawkdovevar/model.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(
+        self,
+        grid_size,
+        include_diagonals=True,
+        hawk_odds=0.5,
+    ):
+        super().__init__(
+            grid_size, include_diagonals=include_diagonals, hawk_odds=hawk_odds
+        )
+        # no custom logic or params yet, but will be adding risk updating logic
```
Comment (quadrismegistus) [inline simulatingrisk/hawkdovevar/model.py:30]: I really like the subclassed approach. It feels more intuitive to read, understand, and probably use/code/maintain

File: tests/test_hawkdove.py [hunk 3]
```diff
@@ -31,66 +41,78 @@ def test_agent_initial_choice():
...
     mock_neighbors = [
         Mock(last_choice=Play.HAWK),
         Mock(last_choice=Play.HAWK),
         Mock(last_choice=Play.HAWK),
         Mock(last_choice=Play.DOVE),
     ]
 
-    with patch.object(HawkDoveAgent, "neighbors", mock_neighbors):
+    with patch.object(HawkDoveSingleRiskAgent, "neighbors", mock_neighbors):
         assert agent.num_dove_neighbors == 1
```
Comment (quadrismegistus) [inline tests/test_hawkdove.py:111]: These are some really nice tests. Are any of them 'integration' tests would you say (making sure results match expectation from theory/math, e.g.) or are these mainly unit tests (making sure code does what it says it does)?

**Review by quadrismegistus — APPROVED** [review]
Looks great!

File: simulatingrisk/batch_run.py (hunk 1 above)
Comment (rlskoeser) [reply simulatingrisk/batch_run.py:70]: I'm not certain I need to store `risk_attitudes` anymore strictly speaking but there was one chart that used it to check how to display and I didn't refactor that out yet, so I kept it.

File: simulatingrisk/hawkdovevar/model.py (hunk 2 above)
Comment (rlskoeser) [reply simulatingrisk/hawkdovevar/model.py:30]: thank you, really glad to hear that - the could already feels easier to work with to me for expanding the options on the variable version without messing up the single risk version

File: tests/test_hawkdove.py (hunk 3 above)
Comment (rlskoeser) [reply tests/test_hawkdove.py:111]: I think of integration tests as testing interactions between different systems, like tests in some of our other apps that check that solr results return what we expect for a given set of test input in the solr index. I don't think I've written anything here that isn't a unit test, but the ones that have probabilistic results are a little weird and more mathy, maybe...

**Pull request merged**